src/Analyser/TypeSpecifier.php

```diff
@@ -674,59 +675,75 @@ public function specifyTypesInCondition(
 			&& count($expr->vars) > 0
 			&& !$context->null()
 		) {
-			if (!$context->true()) {
-				if (!$scope instanceof MutatingScope) {
-					throw new ShouldNotHappenException();
-				}
-
-				$specifiedTypes = new SpecifiedTypes();
+			// rewrite multi param isset() to and-chained single param isset()
+			if (count($expr->vars) > 1) {
```

Comment:
the actual change is here.

Implemented the idea expressed in https://github.com/phpstan/phpstan/issues/8366#issuecomment-1314755824